python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -563,12 +545,7 @@ class RandomForestClassifier(Base):
         concat_model_handle = concatenate_trees(deref(model_handles))
         cdef uintptr_t concat_model_ptr = <uintptr_t> concat_model_handle
         self.treelite_handle = concat_model_ptr
-
-        cdef vector[unsigned char] pbuf_mod_info = \
-            save_model(<ModelHandle> concat_model_ptr)
-        cdef unsigned char[::1] pbuf_mod_view = \
-            <unsigned char[:pbuf_mod_info.size():1]>pbuf_mod_info.data()
-        self.model_pbuf_bytes = bytearray(memoryview(pbuf_mod_view))
+        self.model_serialized = treelite_serialize(concat_model_ptr)
```

Comment:
Does this get cleaned up in `__del__` correctly? I.e. does it do any allocation in C++ or is it all in normal python objects?

Reply:
There is no new allocation at all. The trees are already in memory and we perform essentially a pointer cast to get byte representation.